c++/src/kj/compat/http.c++

```diff
@@ -2703,12 +2783,205 @@ private:
+      for (const auto& part : parts) {
+        memcpy(&processedMessage[currentIndex], part.buffer.begin(), part.size);
+        // We need to use `part.size` to determine the number of useful bytes, since data after
+        // `part.size` is unused (and probably junk).
+        currentIndex += part.size;
+      }
+
+      if (addNullTerminator) {
+        processedMessage[currentIndex++] = '\0';
+      }
+
+      KJ_ASSERT(currentIndex == processedMessage.size());
+
+      return kj::mv(processedMessage);
+    }
+
+    void reset() {
+      // Resets the (de)compression context. This should only be called when the (de)compressor uses
+      // client/server_no_context_takeover.
+      switch (mode) {
+        case Mode::COMPRESS: {
+          KJ_ASSERT(deflateReset(&ctx) == Z_OK, "deflateReset() failed.");
+          break;
+        }
+        case Mode::DECOMPRESS: {
+          KJ_ASSERT(inflateReset(&ctx) == Z_OK, "inflateReset failed.");
+          break;
+        }
+      }
+
+    }
+
+  private:
+    Result pumpOnce() {
+      // Prepares Zlib's internal state for a call to deflate/inflate, then calls the relevant
+      // function to process the input buffer. It is assumed that the caller has already set up
+      // Zlib's input buffer.
+      //
+      // Since calls to deflate/inflate will process data until the input is empty, or until the
+      // output is full, multiple calls to `pumpOnce()` may be required to process the entire
+      // message. We're done processing once either `result` is `Z_STREAM_END`, or we get
+      // `Z_BUF_ERROR` and did not write any more output.
+      size_t bufSize = 4096;
+      Array<kj::byte> buffer = kj::heapArray<kj::byte>(bufSize);
+      ctx.next_out = buffer.begin();
+      ctx.avail_out = bufSize;
+
+      int result = Z_OK;
+
+      switch (mode) {
+        case Mode::COMPRESS:
+          result = deflate(&ctx, Z_SYNC_FLUSH);
+          KJ_REQUIRE(result == Z_OK || result == Z_BUF_ERROR || result == Z_STREAM_END,
+                      "Compression failed.");
+          break;
+        case Mode::DECOMPRESS:
+          result = inflate(&ctx, Z_SYNC_FLUSH);
+          KJ_REQUIRE(result == Z_OK || result == Z_BUF_ERROR || result == Z_STREAM_END,
+                      "Decompression failed.");
+          break;
+      }
+
+      return Result {
+        result,
+        kj::mv(buffer),
+        bufSize - ctx.avail_out,
+      };
+    }
+
+    kj::Vector<Result> processLoop(kj::Maybe<size_t> maxSize) {
+      // Since Zlib buffers the writes, we want to continue processing until there's nothing left.
+      kj::Vector<Result> output;
+      size_t totalBytesProcessed = 0;
+      for (;;) {
+        Result result = pumpOnce();
+
+        auto status = result.processResult;
+        auto bytesProcessed = result.size;
+        if (bytesProcessed > 0) {
+          output.add(kj::mv(result));
+          totalBytesProcessed += bytesProcessed;
+          KJ_IF_MAYBE(maxMsgSize, maxSize) {
```

Comment:
In this `KJ_IF_MAYBE`, the unwrapped name `maxMsgSize` is a variation of the wrapped name `maxSize` that doesn't really express what the difference is in a meaningful way.

I haven't come up with a great answer for what to do in these situations, but I don't like these random variations. Instead, I've mostly settled on using a single-letter name for the unwrapped version. So here I'd do `KJ_IF_MAYBE(m, maxSize)`. It's not beautiful but at least by being a consistent convention across the codebase, it's easy for readers to understand what they're reading.

Reply:
https://github.com/capnproto/capnproto/pull/1505/commits/c9f2965ffc7a2db2d0989f45520a4f17df25003f